src/book/02-system/02-subsystems/04-motion.mdx

```diff
@@ -5,3 +5,66 @@ title: Motion
+
+This walk is based on the work of [Yi et al.](https://doi.org/10.1109/HUMANOIDS.2013.7029960) and Song ('Development of an Omni-directional Gait Generator and a Stabilization Feedback Controller for Humanoid Robots'). The code is a refactored version of the [UPennalizers](https://fling.seas.upenn.edu/~robocup/wiki/) / Team DARwIn [2013 code release](https://github.com/UPenn-RoboCup/UPennalizers/tree/master). NUbots have made several improvements and simplifications.
+
+The module for this walk is located in the [OldWalkEngine module folder](https://github.com/NUbots/NUbots/tree/master/module/motion/OldWalkEngine).
+
+## Quintic Spline Walk
+
+The Quintic Walk is an **open loop controller** based on **quintic splines** (piecewise fifth degree polynomials). It sets points in each direction for rotation and translation and uses those points to create smooth quintic splines.
+
+The engine has support for kicking, however this is not used at this time.
+
+The code used for this walk comes from [**BitBots**](https://github.com/bit-bots/bitbots_motion/), which in turn was based off [Quentin "Leph" Rouxel and Team Rhoban's walk](https://github.com/Rhoban/model/).
+
+The implementation of this walk is currently a work in progress. See the implementation on the [QuinticWalk branch](https://github.com/NUbots/NUbots/tree/biddulph/quintic_walk/module/motion/QuinticWalk).
+
+### Implementation
+
+The walk uses the Footstep class to generate the targets for the foot. It handles the special cases of side stepping and turning to prevent collisions with the other foot.
+
+The SplineContainer class holds Points and multiple Spline objects. As the SplineContainer gets points, it creates Splines based on these points. These Spline objects are matched to some name in a mapping.
+
+The Spline that the SplineContainer holds is a generic Spline class that holds all the information for a mathematical spline. Each piecewise polynomial in the spline is represented by a Polynom object that contains the coefficients for a polynomial.
+
+The specific Spline type we use is a SmoothSpline, extended from the Spline class. Find out more about splines and this specific type of spline on the [mathematics](/system/foundations/mathematics#splines) page.
+
+The TrajectoryUtils file specifies the usage of the SplineContainer for the purpose of walking. It creates a SplineContainer with the following Splines
+
+| Name                 | Description                                                                              |
+| -------------------- | ---------------------------------------------------------------------------------------- |
+| IS_DOUBLE_SUPPORT    | Splines for when the robot is in double support phase, i.e. both feet are on the ground. |
+| IS_LEFT_SUPPORT_FOOT | Splines for the support foot.                                                            |
```

Comment:
Could be a dumb question, but do you know why they have defined LEFT_SUPPORT and not right too? Methinks that seems quite strange, looking at how that class uses that enum

Reply:
It's because the name references a boolean which is uses when it makes the splines. It is a weird thing to call it, but when it makes the splines one of the parameters is based on if this is the left foot or not. 
`point(TrajectoryTypes::IS_LEFT_SUPPORT_FOOT, 0.0f, foot_step.isLeftSupport());`

Reply:
Oh it is a bool? I misunderstood when I was looking then. Too easy.